samples/modules/thrift/hello_client/src/main.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2022 Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifdef __ZEPHYR__
+#include <zephyr/kernel.h>
+#endif
+
+#include <cstdlib>
+#include <iostream>
+
+#include <fcntl.h>
+
+#include <thrift/protocol/TBinaryProtocol.h>
+#include <thrift/protocol/TCompactProtocol.h>
+#include <thrift/transport/TBufferTransports.h>
+#include <thrift/transport/TSSLSocket.h>
+#include <thrift/transport/TSocket.h>
+#include <thrift/transport/TZlibTransport.h>
+
+#include "Hello.h"
+
+using namespace apache::thrift;
+using namespace apache::thrift::protocol;
+using namespace apache::thrift::transport;
+
+#ifndef IS_ENABLED
+#define IS_ENABLED(flag) flag
+#endif
+
+#ifndef CONFIG_THRIFT_COMPACT_PROTOCOL
+#define CONFIG_THRIFT_COMPACT_PROTOCOL 0
+#endif
+
+#ifndef CONFIG_THRIFT_ZLIB_TRANSPORT
+#define CONFIG_THRIFT_ZLIB_TRANSPORT 0
+#endif
+
+#ifndef CONFIG_THRIFT_SSL_SOCKET
+#define CONFIG_THRIFT_SSL_SOCKET 0
+#endif
+
+#ifdef __ZEPHYR__
+int main(void)
+#else
+int main(int argc, char** argv)
+#endif
+{
+  std::string my_addr;
+
+#ifdef __ZEPHYR__
+  my_addr = CONFIG_NET_CONFIG_PEER_IPV4_ADDR;
+#else
+  if (argc >= 2) {
+    my_addr = std::string(argv[1]);
+  } else {
+    my_addr = "192.0.2.1";
+  }
+#endif
+
+  int port = 4242;
+  std::shared_ptr<TSSLSocketFactory> socketFactory;
+  std::shared_ptr<TTransport> trans;
+  if (IS_ENABLED(CONFIG_THRIFT_SSL_SOCKET)) {
+    const int port = 4242;
+    socketFactory = std::make_shared<TSSLSocketFactory>();
+    socketFactory->authenticate(true);
+#ifdef __ZEPHYR__
+    socketFactory->loadCertificateFromBuffer(
+#include "../../hello_common/qemu-cert.pem"
```

Comment:
I was not aware you could even do this, is this a new C++ standard feature? Does this correctly detect and rebuild the application if you build it once, then manually change the contents of the pem file and build it again?

Reply:
It's and old C preprocessor feature. Including any file has always been possible.

Reply:
It's only working because the files have been formatted :) needs to be adapted to work the same way this sample works https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/net/sockets/http_client so that the .pem files are valid .pem files

Reply:
Yes. That is the same request I made to our GSoC contributor. See comment below.